services/s3/src/main/java/software/amazon/awssdk/services/s3/S3Utilities.java

```diff
@@ -215,6 +217,10 @@ private SdkHttpFullRequest createMarshalledRequest(GetUrlRequest getUrlRequest,
         // encode key
         builder.encodedPath(PathMarshaller.GREEDY.marshall(builder.encodedPath(), "Key", getUrlRequest.key()));
 
+        if (StringUtils.isNotBlank(getUrlRequest.versionId())) {
+            builder.appendRawQueryParameter("versionId", getUrlRequest.versionId());
+        }
```

Comment:
It looks like our default marshalling for GetObjectRequest wouldn't treat non-null differently for strings. Should we just do getUrlRequest.versionId() != null to be consistent with how GetObjectRequestMarshaller works?

Reply:
Yeah, makes sense. Updated